added edit faq function

kozr opened the pull request:
I've actually done this more than few days ago but forgot to commit and push lol

kozr committed:
added edit faq function

Eric1015 commented on components/FAQ.vue:
```diff
@@ -0,0 +1,98 @@
+      {{ faq.data.answer }}
+      <br />
+      Timestamp:
+      {{ faq.data.timestamp }}
```
Just curious what will be displayed here when timestamp is null?
I can't see it in data object, and it seems like "firebaseTimestamp" will be the key that assigned to the data (if i read ...firebaseTimestamp properly).

kozr replied:
forgot to merge branch earlier, it should be clear now. (pressing save will add a firebaseTimestamp)

Eric1015 commented:
Also will be easier for us to review if there can be a screen shot.

Ciuffi commented:
Nice! 

Could we also save the user who made edits to that faq question? I think you can just use the firebase auth module to grab the id.

kozr commented:
omg my changes in index.vue AND firebase.js didn't save. Aaaaaaaaaaaa
Edit: nvm I was on a different branch

kozr committed:
merged dev into faqSetup

kozr committed:
Merge branch 'dev' into faqSetup

kozr committed:
added add and remove faq

kozr committed:
Merge branch 'faqSetup' of https://github.com/nwplus/cms into faqSetup

kozr committed:
fixed bugs and added last edited

kozr commented:
I've added "last edited by", and added "add faq" and "delete faq" and also a bunch of template functions.
Normal:
![image](https://user-images.githubusercontent.com/40312631/63839013-4c2c5680-c933-11e9-95fd-da1666668824.png)

After clicking on nwplus button:
![image](https://user-images.githubusercontent.com/40312631/63838976-40d92b00-c933-11e9-9351-3f9258b2bbbb.png)

After clicking on edit:
![image](https://user-images.githubusercontent.com/40312631/63839083-71b96000-c933-11e9-9dc3-c27d084696d4.png)

Ciuffi commented on components/FAQ.vue:
```diff
@@ -0,0 +1,131 @@
+      this.data.question = 'Sample question'
+      this.data.category = 'Sample category'
+      this.data.answer = 'Sample answer'
+      this.$emit('refreshData')
```
What does this line do?

Ciuffi replied:
Specifically the `$emit` line

kozr replied:
The emit tells the parent (cms.vue) to run the method "refreshData"

Ciuffi replied:
ahh very cool

Ciuffi commented on plugins/firebase.js:
```diff
@@ -78,6 +78,24 @@ const fireDb = {
       .doc(docId)
       .update(object)
+  },
+  add: async (webDocument, collection, object) => {
```
Well done with these methods!

Ciuffi commented on components/FAQ.vue:
```diff
@@ -0,0 +1,131 @@
+    handleEdit(faq) {
+      this.editMode = !this.editMode
+      this.currentFaq = faq
+      this.data.selected = faq.data.selected
```
What does selected do? Can you select multiple?

kozr replied:
It's to decide if you want to display that specific faq

Ciuffi replied:
what are the check boxes for?

kozr replied:
on LHD_2020, there is a v-if conditional rendering statement that checks if a specific faq is selected. If it is true, then it'll display that faq, otherwise it won't display the specific faq. I'll leave some comments next time mb._

kozr replied:
Each checkbox corresponds to each individual faq. If it's checked, it'll display on the website, else it wouldn't

Ciuffi replied:
ahhh very nice, thanks for clarifying!

Ciuffi reviewed (commented):
I just have some small questions, Otherwise it looks really good.

Eric1015 commented on components/FAQ.vue:
```diff
@@ -0,0 +1,131 @@
+        console.log('updated doc')
+        fireDb.update(this.website, 'Faq', this.currentFaq.id, this.data)
+      } else {
+        console.log('added new doc')
```
Maybe you don't  need 'console.log' anymore?

Eric1015 commented on components/FAQ.vue:
```diff
@@ -0,0 +1,131 @@
+      const user = auth.currentUser.displayName
+      this.data = { ...this.data, ...firebaseTimestamp, user }
+      if (this.currentFaq != null && this.currentFaq.id !== undefined) {
+        console.log('updated doc')
```
Maybe you don't need 'console.log' anymore?

Eric1015 commented on components/FAQ.vue:
```diff
@@ -0,0 +1,131 @@
+      this.editMode = !this.editMode
+      if (this.currentFaq != null && this.currentFaq.id !== undefined) {
+        fireDb.delete(this.website, 'Faq', this.currentFaq.id)
+      }
```
Probably want to let the user know that the delete has not been done if this 'if statement' is not  executed

kozr replied:
I believe the only time this.currentFaq or this.currentFaq.id will be null is when the user clicks on "Add FAQ" and immediately delete it before saving. In that case, it shouldn't need to notify the user because it was never uploaded to firebase in the first place (and thus nothing to be deleted). So I don't think it needs an else {alert()} or anything like that. Sorry for the confusion.

Eric1015 replied:
Oh, that makes sense. Thank you for clarifying it!

kozr committed:
removed debugging lines

Eric1015 commented:
![image](https://user-images.githubusercontent.com/31051640/63913125-ee4e5a80-ca69-11e9-8e8f-199a425f8424.png)

Ciuffi committed:
Merge branch 'dev' into faqSetup

Ciuffi committed:
add faq component and refreshData on mount.

Ciuffi reviewed (approved):
Looks good! I just added a call to refreshData in the mounted function of the FAQ Component because the faq questions weren't being loaded on page load.

Take a look at my changes and feel free to merge in your branch when you're ready!
